lib/ruby_lsp/requests/document_link.rb

```diff
@@ -63,6 +101,61 @@ def gem_paths
             lookup
           end, T.nilable(T::Hash[String, T::Array[String]]))
         end
+
+        sig { returns(T.nilable(T::Hash[String, T::Hash[Symbol, String]])) }
+        def rails_documents
```

Comment:
I personally don't think that this functionality belongs in the `DocumentLink` class. How would folks feel about creating a `RailsDocumentsClient` class or something along those lines that can wrap the calls to the Rails documentation and then expose them to the LSP via an API? It would make the code in this class cleaner and then we could even write separate tests for the client to make sure it has good coverage.

(cc: @vinistock would also love your opinion)

Reply:
AND maybe as we expand the functionality of document links we'll have more of these clients and they can share behavior?